### PR TITLE
react-instantsearch-core: add queryID? to SearchResults

### PR DESCRIPTION
**Pull request by tkrugg**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/algolia/react-instantsearch/issues/3085#issuecomment-1027170457
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-instantsearch-core ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-instantsearch-core@latest/))
was missing the following properties:
1. instantSearchContext
1. InstantSearchConsumer
1. InstantSearchProvider
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4c5db099eaeeaf15bc4691b1cd99ff46f8e08028
</p>

File: types/react-instantsearch-core/index.d.ts [hunk 1]
```diff
@@ -694,6 +694,7 @@ export interface SearchResults<TDoc = BasicDoc> {
   facets: any[];
   aroundLatLng?: string | undefined;
   automaticRadius?: string | undefined;
+  queryID?: string;
```
Comment (Haroenv) [inline types/react-instantsearch-core/index.d.ts:697]: should this maybe be imported from algoliasearch in a separate PR?

**Review by Haroenv — APPROVED** [review]

**Comment by Haroenv** [comment]
Ready to merge
